sympy/polys/ring_series.py

```diff
@@ -591,6 +740,7 @@ def _exp1(p, x, prec):
         p1 += tmp
     return p1
 
+@check_precision
```

Comment:
I don't think the check_precision should be called like this, because it is slow (needs to determine the degree). Rather, the user should provide it.